core/comp_ui.py

```diff
@@ -32,18 +32,23 @@
 def _PromptLen(prompt_str):
   """Ignore all characters between \x01 and \x02 and handle unicode characters.
   In particular, the display width of a string may be different from either the
-  number of bytes or the number of unicode characters."""
+  number of bytes or the number of unicode characters.
+  Additionally, if there are multiple lines in the prompt, only give the length of the last line."""
   escaped = False
   display_str = ""
-  for c in prompt_str:
+  last_line = prompt_str.split('\n')[-1]
+  for c in last_line:
     if c == '\x01':
       escaped = True
     elif c == '\x02':
       escaped = False
     elif not escaped:
       display_str += c
   try:
-    return libc.wcswidth(display_str)
+    width = libc.wcswidth(display_str)
+    if width == -1:
+        return len(display_str)
```

Comment:
Please move this outside the try block and use 2-space indent.

IIRC PEP 8 says somewhere to minimize the scope of try blocks.

(It would be nice to set up `test/lint.sh` to check indents)